### ISSUE TITLE
Selection hotkeys vs ranks

### ISSUE THREAD

**Issue by sprunk** [ZeroK-RTS/Zero-K#4802]
https://zero-k.info/Forum/Thread/36004

> I built 7 Athenas, lined them up and set the 2nd, 3rd and 6th to selection rank 0.
Now I pressed my hotkey, the arrow down key, and all of them got selected.
While pressing ctrl+down or alt+down nothing happened.

Perhaps the down key just conflicts with camera movement?

**Comment by GoogleFrog** [comment]
I am pretty sure that the Ctrl exception is literally "When Ctrl is held", so it won't work for selection keys that don't use Ctrl.

**Comment by sprunk** [comment]
Yes, I meant the general hotkey framework regardless of ranks. He said he set his hotkey to just "Down". This can become interpreted by the framework as one of two things in the technical implementation:
 * `Any+Down`, which would mean the "select Athenas" action happens if you hold either `Ctrl+Down` or just `Down` (among others). Then, the selection exception would filter it down if it was Ctrl+Down, or leave it be if it was plain Down.
 * just `Down`, which would mean the "select Athenas" action would not happen at all with `Ctrl` because without the `Any+` that's only matched by Down when Ctrl is not held.

The second interpretation seems to match the "nothing gets selected with Ctrl" part of the report, but it's somewhat unintuitve. On the other hand doing it the other way would likely be worse (for example if the hotkey framework was setting the fight-move hotkey to `Any+A` when somebody just pressed `A` then perhaps `Ctrl+A` would match it and you would have a conflict with the "select everything" hotkey). I don't know how to approach this well.

**Comment by GoogleFrog** [comment]
The hotkey setting UI doesn't have all the power of editing the config file directly. Eg, cycling order can't be set ingame. Perhaps the distinction between Down and Any+Down is another thing that requires file editing. Otherwise what are we supposed to do, have some global toggle that changes how new hotkeys are set (whether Down is interpreted as Any+Down at the point of setting the hotkey).
